src/multicellular_stage/editor/CellBodyPlanEditorComponent.cs

```diff
@@ -979,11 +1081,120 @@ private void ClearSelectedAction()
         EmitSignal(SignalName.OnCellTypeToEditSelected, default(Variant), false);
     }
 
+    private void OnEnergyBalanceOptionsChanged()
+    {
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
+    private void OnResourceLimitingModeChanged(int index)
+    {
+        balanceMode = (ResourceLimitingMode)index;
+
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
     private void OnCellsChanged()
     {
         UpdateAlreadyPlacedVisuals();
 
+        UpdateStats();
+
         UpdateArrow();
+
+        UpdateFinishButtonWarningVisibility();
+    }
+
+    private void UpdateStats()
+    {
+        organismStatisticsPanel.UpdateStorage(GetAdditionalCapacities(out var nominalCapacity), nominalCapacity);
+        organismStatisticsPanel.UpdateSpeed(CellBodyPlanInternalCalculations.CalculateSpeed(editedMicrobeCells));
+        organismStatisticsPanel.UpdateRotationSpeed(CellBodyPlanInternalCalculations.CalculateRotationSpeed(
+            editedMicrobeCells));
+
+        CalculateEnergyAndCompoundBalance(editedMicrobeCells);
+    }
+
+    /// <summary>
+    ///   Calculates the energy balance and compound balance for a colony
+    /// </summary>
+    private void CalculateEnergyAndCompoundBalance(IReadOnlyList<HexWithData<CellTemplate>> cells,
+        BiomeConditions? biome = null)
+    {
+        biome ??= Editor.CurrentPatch.Biome;
+
+        bool moving = organismStatisticsPanel.CalculateBalancesWhenMoving();
+
+        IBiomeConditions conditionsData = biome;
+
+        if (balanceMode != ResourceLimitingMode.AllResources)
+        {
+            conditionsData = new BiomeResourceLimiterAdapter(balanceMode, conditionsData);
+        }
+
+        var energyBalance = new EnergyBalanceInfo();
+        energyBalance.SetupTrackingForRequiredCompounds();
+        foreach (var hex in cells)
+        {
+            ProcessSystem.ComputeEnergyBalance(hex.Data!.Organelles, conditionsData, hex.Data.MembraneType, moving,
```

Comment:
I guess it would get pretty complicated to change but I'd at least add a TODO about the potential to also improve this code's performance (if needed, things are probably pretty fast with 20 or fewer cells in the editor) by calculating things per type.